### ISSUE TITLE
[Documentation] Streamline release process

### ISSUE THREAD

**Issue by okdas** [pokt-network/poktroll#622]
## Objective

`shannon` already has automation to build binaries and production-grade container images for releases.

The objective is to add documentation for the protocol team to cut & publish new releases.

## Origin Document

- The `poktroll` releases can be found [here](https://github.com/pokt-network/poktroll/releases/tag/v0.0.2)
- The `smt` has an example of Makefile helpers and documentation that can be used as a reference:
	- https://github.com/pokt-network/smt/blob/main/Makefile#L97
	- https://github.com/pokt-network/smt?tab=readme-ov-file#release-tags

## Goals

- Streamline the release process
- Unblock protocol team from publishing releases and pre-releases often

## Non-goals

- Handling on-chain upgrades & features switches
- Protocol upgrade playbook

## Deliverables

- [ ] Add a copy-paste template for release announcements.
	- [ ] [TBD] Verify if any of this should be automated by Github Action.
- [ ] Document the process and post instructions on `dev.poktroll.com` for internal use by the protocol team.
- [ ] Add the appropriate Makefile helpers

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @okdas 
**Co-owner**: @Olshansk

**Comment by Olshansk** [comment]
@okdas I updated some details and deliverables. Will be creating another ticket dedicated to protocol upgrades

**Comment by okdas** [comment]
OK this has been moving for a couple of iterations now while technically being 90% done. I'll pick it up today to finish the documentation (most of which is already merged) and add the Makefile target.
